Exiled.Events/Patches/Events/Scp914/InteractingEvents.cs

```diff
@@ -16,51 +19,93 @@ namespace Exiled.Events.Patches.Events.Scp914
+                new(OpCodes.Bne_Un_S, ifLabel),
+                new(OpCodes.Ldc_I4_0),
+                new(OpCodes.Stloc_0),
+                new(OpCodes.Br_S, elseLabel),
+                new CodeInstruction(OpCodes.Ldc_I4_1).WithLabels(ifLabel),
+                new(OpCodes.Ldarg_0),
+                new(OpCodes.Ldfld, Field(typeof(Scp914Controller), nameof(Scp914Controller._knobSetting))),
+                new(OpCodes.Add),
+                new(OpCodes.Stloc_0),
+                new CodeInstruction(OpCodes.Ldarg_1).WithLabels(elseLabel),
+                new(OpCodes.Call, Method(typeof(Player), nameof(Player.Get), new[] { typeof(ReferenceHub) })),
+                new(OpCodes.Ldloc_0),
+                new(OpCodes.Ldc_I4_1),
+                new(OpCodes.Newobj, GetDeclaredConstructors(typeof(ChangingKnobSettingEventArgs))[0]),
+                new(OpCodes.Dup),
+                new(OpCodes.Dup),
+                new(OpCodes.Stloc_S, changingKnobSetting.LocalIndex),
+                new(OpCodes.Callvirt, Method(typeof(Handlers.Scp914), nameof(Handlers.Scp914.OnChangingKnobSetting))),
+                new(OpCodes.Callvirt, PropertyGetter(typeof(ChangingKnobSettingEventArgs), nameof(ChangingKnobSettingEventArgs.IsAllowed))),
+                new(OpCodes.Brfalse_S, returnLabel),
+                new(OpCodes.Ldarg_0),
+                new(OpCodes.Ldloc_S, changingKnobSetting.LocalIndex),
+                new(OpCodes.Callvirt, PropertyGetter(typeof(ChangingKnobSettingEventArgs), nameof(ChangingKnobSettingEventArgs.KnobSetting))),
+                new(OpCodes.Callvirt, PropertySetter(typeof(Scp914Controller), nameof(Scp914Controller.Network_knobSetting))),
+            });
+
+            offset = -3;
+            index = newInstructions.FindLastIndex(instruction => instruction.opcode == OpCodes.Stfld && (FieldInfo)instruction.operand == Field(typeof(Scp914Controller), nameof(Scp914Controller._remainingCooldown))) + offset;
+
+            newInstructions.InsertRange(index, new[]
+            {
+                new CodeInstruction(OpCodes.Ldarg_1).MoveLabelsFrom(newInstructions[index]),
+                new(OpCodes.Call, Method(typeof(Player), nameof(Player.Get), new[] { typeof(ReferenceHub) })),
+                new(OpCodes.Ldc_I4_1),
+                new(OpCodes.Newobj, GetDeclaredConstructors(typeof(ActivatingScp914EventArgs))[0]),
+                new(OpCodes.Dup),
+                new(OpCodes.Callvirt, Method(typeof(Handlers.Scp914), nameof(Handlers.Scp914.OnActivating))),
+                new(OpCodes.Callvirt, PropertyGetter(typeof(ActivatingScp914EventArgs), nameof(ActivatingScp914EventArgs.IsAllowed))),
+                new(OpCodes.Brfalse_S, returnLabel),
+            });
+
+            newInstructions[newInstructions.Count - 1].labels.Add(returnLabel);
+
+            for (int z = 0; z < newInstructions.Count; z++)
+            {
+                Log.Debug(newInstructions[z].opcode + " " + newInstructions[z].operand);
```

Comment:
You forgot to delete those debug lines (Joker 2.0?)

Reply:
Oh, yes) i remove it soon